docs/usage/MTCAPIDoc.md

```diff
@@ -0,0 +1,417 @@
+
+- spec.migPlanRef: ObjectReference contains enough information to let you inspect or modify the referred object. This is a required field.
+
+## Mig Cluster
+
+Local Mig cluster: 
+```
+ 
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigCluster
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: migcluster-local
+  namespace: openshift-migration
+spec:
+  # [!] Change isHostCluster to 'false' if you want to use a clusterRef and serviceAccountSecretRef
+  #     instead of using the mig-controller detected kubeconfig. Refer to mig-cluster-aws.yaml for an example.
+  isHostCluster: true
+
+  # [!] Change refresh to 'true' to force a manual reconcile
+  refresh: false
+  ```
+
+  Remote Mig cluster:
+
+  ```
+  apiVersion: migration.openshift.io/v1alpha1
+kind: MigCluster
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: migcluster-remote
+  namespace: openshift-config
+spec:
+  # [!] Change isHostCluster to 'true' if you'll be running the controller on this cluster.
+  #     Setting 'isHostCluster' to true will bypass using the clusterRef and serviceAccountSecretRef below.
+  isHostCluster: false
+
+  url: "https://my-remote-cluster-ip.nip.io:8443"
+
+  serviceAccountSecretRef:
+    name: sa-token-remote
+    namespace: openshift-config
+
+  # [!] Change refresh to 'true' to force a manual reconcile
+  refresh: false
+  ```
+- spec.azureResourceGroup: For azure clusters -- it's the resource group that in-cluster volumes use.
+
+- spec.caBundle: If the migcluster needs SSL verification for connections a user can supply a custom CA bundle. This field is required only when spec.Insecure is set false.
+  
+- spec.exposedRegistryPath: Stores the path of registry route when using direct migration.
+
+- spec.insecure: If set false, user will need to provide CA bundle for TLS connection to the remote cluster.
+
+- spec.isHostCluster: Specifies if the cluster is host (where the controller is installed) or not. This is a required field.
+
+- spec.refresh: If set True, forces the controller to run a full suite of validations on migcluster.
+
+- spec.restartRestic: An override setting to tell the controller that the source cluster restic needs to be restarted after stage pod creation.
+
+- spec.serviceAccountSecretRef: ObjectReference contains enough information to let you inspect or modify the referred object.
+
+- spec.url: Stores the url of the remote cluster. The field is only required for the source cluster object.
+
+## MigHooks
+
+```
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigHook
+metadata:
+  generateName: test-
+  name: test-btmv5
+  namespace: openshift-migration
+spec:
+  custom: false
+  image: 'quay.io/konveyor/hook-runner:latest'
+  playbook: >-
+    LSBuYW1lOiBTbGVlcCBmb3IgMTAgc2Vjb25kcyBhbmQgY29udGludWUgd2l0aCBwbGF5CiAgd2FpdF9mb3I6CiAgIadsfwerq0OiAxMA==
+  targetCluster: source
+```
+
+- spec.activeDeadlineSeconds: Specifies the highest amount of time for which the hook will run.
```

Comment:
@eriknelson @shubham-pampattiwar 

spec.activeDeadlineSeconds: What is the default value? Is there a max value?

Reply:
@apinnick if the value for this field is not provided then there will not be any limit for which the hook should run, it would be indefinite.

Reply:
@JaydipGabani That is true for jobs, but we set a default of 1800 seconds (30 minutes) so we don't ever infinitely hang a migration. https://github.com/konveyor/mig-controller/blob/b20990b61d1c1b8c61df3f655c3fffb83ff6bd70/pkg/controller/migmigration/hooks.go#L260-L264

It's stored as an int64 value. I've never tried anything up to that value, but an int64 max value is `9223372036854775807`, basically infinite.